src/core/FileWatcher.cpp

```diff
@@ -166,21 +166,23 @@ void BulkFileWatcher::addPath(const QString& path)
         const bool directorySuccess = m_fileWatcher.addPath(directoryPath);
         m_watchedPaths[directoryPath] = directorySuccess;
     }
-    m_watchedFilesInDirectory[directoryPath][filePath] = info.exists();
+    m_watchedFilesInDirectory[directoryPath][filePath] = info.exists() ? info.lastModified().toMSecsSinceEpoch() : 0;
 }
 
 void BulkFileWatcher::handleFileChanged(const QString& path)
 {
     const QFileInfo info(path);
     const QString filePath = info.absoluteFilePath();
     const QString directoryPath = info.absolutePath();
-    const QMap<QString, bool>& watchedFiles = m_watchedFilesInDirectory[directoryPath];
-    const bool created = !watchedFiles[filePath] && info.exists();
-    const bool deleted = watchedFiles[filePath] && !info.exists();
-    const bool changed = !created && !deleted;
+    const QMap<QString, qint64>& watchedFiles = m_watchedFilesInDirectory[directoryPath];
+    const qint64 lastModificationTime = info.lastModified().toMSecsSinceEpoch();
+    const bool created = 0 == watchedFiles[filePath] && info.exists();
+    const bool deleted = 0 != watchedFiles[filePath] && !info.exists();
```

Comment:
I prefer to wrap in-line boolean settings with parenthesis. Also put the static value second (ie, ```const bool created = (watchedFiles[filePath] == 0 && info.exists());```